python/src/trezorlib/ethereum.py

```diff
@@ -17,11 +17,157 @@
 from . import exceptions, messages
 from .tools import expect, normalize_nfc, session
 
+from eth_abi.packed import encode_single_packed
+import json
+import re
 
 def int_to_big_endian(value):
     return value.to_bytes((value.bit_length() + 7) // 8, "big")
 
 
+type_name_re = re.compile("^\\w*")
+
+
+def find_typed_dependencies(primary_type: str, types: dict, results: list = None):
+    """
+    Finds all types within a type definition object
+
+    primary_type - Root type
+    types - Type definitions
+    results - Current set of accumulated types
+    """
+    if results is None:
+        results = []
+
+    m = type_name_re.match(primary_type)
+    if m:
+        primary_type = m.string[m.start():m.end()]
```

Comment:
If it's more canonical then OK. I'm from Go background, for us, the explicit code is better. 
The goal is to strip also `[][5]` and whatever it might have on the end. We're interested in the Type name only.

> Can primary type be an empty string?

The type name must be a valid identifier as a name in Solidity. So the answer is no.

Reply:
Yeah, it's how it's usually written. If it can't be empty then let's change the regex to `^\w+`?

Reply:
Changed in a05275a